pymongo/mongo_client.py

```diff
@@ -1467,12 +1528,25 @@ def is_retrying() -> bool:
                 # be a persistent outage. Attempting to retry in this case will
                 # most likely be a waste of time.
                 raise
+            except ConnectionFailure as exc:
+                # This exception is endemic in read-sponsored issues``
+                if not retryable or (retrying and not multiple_retries):
+                    raise
+                retrying = True
+                last_error = exc
+            except OperationFailure as exc:
+                # This exception is endemic in read-sponsored issues``
+                if not retryable or (retrying and not multiple_retries):
+                    raise
+                if exc.code not in helpers._RETRYABLE_ERROR_CODES:
+                    raise
+                retrying = True
+                last_error = exc
```

Comment:
This case is missing the "RetryableWriteError" logic from the PyMongoError case.